app/(api)/_datalib/auth/verifyLogin.ts

```diff
@@ -0,0 +1,40 @@
+import bcrypt from 'bcrypt';
+import jwt, { Secret } from 'jsonwebtoken';
+
+/**
+ * Compares a provided password with a stored password hash.
+ * @param providedPassword The password provided by the user.
+ * @param storedHash The hash stored in the database.
+ * @returns True if the password matches the hash, false otherwise.
+ */
+export async function comparePasswordHash(
+  providedPassword: string,
+  storedHash: string
+): Promise<boolean> {
+  return bcrypt.compare(providedPassword, storedHash);
+}
+
+export async function verifyLogin(email: string, password: string) {
+  try {
+    // If unsuccessful login of user or password, return AuthenticationError.
+    // getUserByEmail TO BE IMPLEMENTED.
+    const user = await getUserByEmail(email);
```

Comment:
For now, u might want to make some dummy getUserByEmail function so it can pass the checks